pkg/alertmanager/amcfg_test.go

```diff
@@ -3826,7 +3827,33 @@ func TestSanitizePagerDutyConfig(t *testing.T) {
 					{
 						PagerdutyConfigs: []*pagerdutyConfig{
 							{
-								Source: "",
+								Source: nil,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:           "Test source is added in pagerduty config for supported versions",
+			againstVersion: semver.Version{Major: 0, Minor: 25},
+			in: &alertmanagerConfig{
+				Receivers: []*receiver{
+					{
+						PagerdutyConfigs: []*pagerdutyConfig{
+							{
+								Source: sourceVal,
```

Comment:
I have kept that to ensure that we are getting the same pointer instance in both in `expect` block as passed in the `in` block. 

Not sure if I am correctly understanding the way these tests work. I wanted to make sure we receive the same pointer instance.

Reply:
Should I go ahead and remove this and replace with `ptr.To("foo")`?

Reply:
@codeknight03  I am fine with either. I was just curious, may be you can test with and without the variable and see if any difference

Also once you address @simonpasquier's comment below this should be good to go I think.